### PR TITLE
[website:docs] rework 'deploy contract' quick-start

### PR DESCRIPTION
**Pull request by jmcook1186**
PR includes a rework of the existing deploy-contract quick start. 

The main change is that it prioritizes deploying on a local blockchain (Anvil) to start, then moves to a public network later.  This allows beginners to quickly get to grips with deploying code without having to find SepoliaETH and with the ability to use ephemeral accounts.

Also adds some detail to help beginners understand what is happening at each stage of the deployment.

### CODE REVIEW

File: docs/src/quickstart/deploy_contract.md [hunk 1]
```diff
@@ -1,33 +1,41 @@
...
 
+> Note: Anvil needs to keep running throughout this tutorial - if you close the terminal your blockchain will cease to exist. Once Anvil has started, open a **new terminal** tab/window to run the rest of the commands in this guide.
 
 ### Making the deployment transaction
 
-Let's recall that we finished our guest book in the previous chapter with the following code.
+In the previous guide you wrote the following contract, and compiled it using `./fe build guest_book.fe --overwrite` to obtain the contract bytecode. This compilation stage converts the human-readable Fe code into a format that can be efficiently executed by Ethereum's embedded computer, known as the Ethereum Virtual Machine (EVM). The bytecode is stored at an address on the blockchain. The contract functions are invoked by sending instructions in a transaction to that address.
 
-```fe
+```rust
```
Comment (cburgdorf) [inline docs/src/quickstart/deploy_contract.md:38]: Note that snippets of `fe` are automatically checked to compile during CI. With Fe being a fast evolving language that slightly breaks compatibility from version to version, this ensures that the examples in our docs keep functioning.

Comment (jmcook1186) [inline docs/src/quickstart/deploy_contract.md:38]:
```suggestion
```fe
```

Comment (jmcook1186) [reply docs/src/quickstart/deploy_contract.md]: ok, thanks

**Comment by cburgdorf** [comment]
@jmcook1186 this is now live https://fe-lang.org/docs/quickstart/deploy_contract.html

**Pull request merged**